### ISSUE TITLE
Review link travel times in BeamPath

### ISSUE THREAD

**Issue by michaz** [LBNL-UCB-STI/beam#934]
The overall leg travel time is in `EmbodiedBeamLeg`, not in `BeamLeg` or in `BeamPath`, because it is not as such part of the plan of the agent but can turn out to be different -- that's why we "re-embody" the legs from old plans when we re-use them.

So I guess the link travel times should be in `EmbodiedBeamLeg`, too, rather than in `BeamPath`? Currently looks to me like they are never updated?

**Comment by michaz** [comment]
Maybe the reason for them being there is that `DrivesVehicle` doesn't have `EmbodiedBeamLegs` but only `BeamLegs` with `BeamPaths`, through the `PassengerSchedule`? Then we should re-visit that, there's no reason why `DrivesVehicle` shouldn't know the full picture..

**Comment by colinsheppard** [comment]
Yes, let's revisit. The whole rationale for maintaining BeamLegs separate from EmbodiedBeamLegs is also worth revisiting. 

Of course it's good to keep the data we need to pass around limited, but much more important is to have data structures that are sensible, easy to understand, easy to develop against.

**Comment by michaz** [comment]
Decided with @colinsheppard to just poke those data structures a bit to see where they can be flattened/simplified/relieved from leftovers of requirements that aren't there anymore.
